bin/varnishncsa/varnishncsa.c

```diff
@@ -1071,17 +1071,25 @@ dispatch_f(struct VSL_data *vsl, struct VSL_transaction * const pt[],
 			case SLT_RespHeader:
 				process_hdr(&CTX.watch_resphdr, b, e);
 				break;
+			case SLT_HitMiss:
+				CTX.hitmiss = "miss";
+				CTX.handling = "hitmiss";
+				break;
+			case SLT_HitPass:
+				CTX.hitmiss = "miss";
+				CTX.handling = "hitpass";
+				break;
```

Comment:
Shouldn't it be the other way around? At this point we set hitmiss to either "hitmiss" or "hitpass" (lookup outcome) and we leave the handling alone, it will be set by the relevant subroutine once we reach it.

Reply:
My understanding is that `varnishncsa` users generally rely on `Varnish:hitmiss` to return either `hit` or `miss` in order to generate statistics and find misses. I would thus think changing this fundamental format would not be advisable.
`Varnish:handling`, however, already has more distinct values, providing more fine grained information on the nature of a hit or miss.

Reply:
FWIW, the format being called `Varnish:hitmiss` (read "hit or miss") is probably not the best choice in light of `hitmiss` as a lookup outcome.

Reply:
If I may: please break stuff

we have releases every 6 months, with full latitude to increment the major version. I don't say we should be as reckless as I'd hope us to be, but changing APIs isn't that big of a deal. Being correct and useful is more important

The current state of clients, as I see it is that `varnishncsa` is only a way to get data out of Varnish before shipping it to a log aggregator. These platform are fully equipped to then search based on regex, prefixes and whatnot. So the only true adjustment would be to update the search query from `== miss` to `== miss*` or similar.

Reply:
@gquintard I do 💯% agree that _being correct and useful is more important_, this is, I hope, what I strive for all the time. And, fwiw, I think my concrete proposal in here does just that: It breaks `handling` by adding two more values.

However, I do not see the advantage of diversifying `hitmiss` over diversifying `handling`, which already _is_ more fine grained. If we have two values, of which one is actually redundant because it only depends on the other, then why should we spread additional information over the two?

Please give me a compelling reason.

Reply:
Let's continue with the proposal in https://github.com/varnishcache/varnish-cache/pull/4217#discussion_r1822170362 that doesn't need to break anything.